.changeset/metal-pigs-share.md

```diff
@@ -0,0 +1,53 @@
+---
+'@shopify/polaris': patch
+---
+
+- Revert "[Internal]: Generate contiguous responsive CSS variables."
+
+  Initially, this was introduced in 3536e3aa0 to work around the following being invalid (`calc()` can't handle `initial`):
+
+  ```css
+  --foo: initial;
+  calc(10px + var(--foo));
+  ```
+
+  However, the only usage so far is in 0f97293b0 where a different work around was used (falling back to a "0" value):
+
+  ```
+  --foo: initial;
+  calc(10px + var(--foo, 0rem));
+  ```
+
+  The other side-effect of introducing contiguous responsive props was excessive inline styles being added to some components. For example;
+
+  ```tsx
+  <Box padding="4" paddingInlineStart="6" paddingBlockEnd="8" />
+  ```
+
+  would result in:
+
+  ```css
+  --pc-box-padding-xs: var(--p-spacing-4);
+  --pc-box-padding-sm: var(--p-spacing-4);
+  --pc-box-padding-md: var(--p-spacing-4);
+  --pc-box-padding-lg: var(--p-spacing-4);
+  --pc-box-padding-xl: var(--p-spacing-4);
+  --pc-box-padding-inline-start-xs: var(--p-spacing-6);
+  --pc-box-padding-inline-start-sm: var(--p-spacing-6);
+  --pc-box-padding-inline-start-md: var(--p-spacing-6);
+  --pc-box-padding-inline-start-lg: var(--p-spacing-6);
+  --pc-box-padding-inline-start-xl: var(--p-spacing-6);
+  --pc-box-padding-block-end-xs: var(--p-spacing-8);
+  --pc-box-padding-block-end-sm: var(--p-spacing-8);
+  --pc-box-padding-block-end-md: var(--p-spacing-8);
+  --pc-box-padding-block-end-lg: var(--p-spacing-8);
+  --pc-box-padding-block-end-xl: var(--p-spacing-8);
+  ```
+
+  Reverting the `makeContiguous` calculation results in simpler inline styles which correctly fall back thanks to `var()`'s handling of `initial`. The same example above becomes:
+
+  ```css
+  --pc-box-padding-xs: var(--p-spacing-4);
+  --pc-box-padding-inline-start-xs: var(--p-spacing-6);
+  --pc-box-padding-block-end-xs: var(--p-spacing-8);
+  ```
```

Comment:
I feel like we should keep the changelog neat and tidy. They'll link to the PR description where this content should live

Reply:
Good call - I've simplified the changelog 👍